src/Adapter/MSTest.TestAdapter/Execution/TestExecutionManager.cs

```diff
@@ -134,6 +135,40 @@ public void RunTests(IEnumerable<TestCase> tests, IRunContext? runContext, IFram
         }
     }
 
+    /// <summary>
+    /// Runs the tests.
+    /// </summary>
+    /// <param name="tests">Tests to be run.</param>
+    /// <param name="runContext">Context to use when executing the tests.</param>
+    /// <param name="frameworkHandle">Handle to the framework to record results and to do framework operations.</param>
+    /// <param name="runCancellationToken">Test run cancellation token.</param>
+    internal async Task RunTestsAsync(IEnumerable<TestCase> tests, IRunContext? runContext, IFrameworkHandle frameworkHandle, TestRunCancellationToken runCancellationToken)
```

Comment:
Note for review: This is an exact copy of the method above, but an async version.